rust/crd/src/lib.rs

```diff
@@ -80,7 +96,9 @@ pub struct ZookeeperClusterSpec {
     #[serde(default = "global_config_default")]
     pub config: GlobalZookeeperConfig,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub servers: Option<Role<ZookeeperConfig>>,
+    pub servers: Option<Role<ZookeeperConfigFragment>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
For auto-generated CRD documentation :rocket: 
```suggestion
    /// Name of the Vector discovery ConfigMap.
    /// It must contain the key `ADDRESS` with the address of the Vector aggregator.
    #[serde(skip_serializing_if = "Option::is_none")]
```